pkg/globalnet/controllers/ipam/ipam.go

```diff
@@ -257,11 +257,17 @@ func (i *Controller) handleUpdateService(old, newObj interface{}) {
 		return
 	}
 
-	if i.excludeNamespaces[newObj.(*k8sv1.Service).Namespace] {
+	service := newObj.(*k8sv1.Service)
+	if i.excludeNamespaces[service.Namespace] {
 		klog.V(log.DEBUG).Infof("In handleUpdateService, skipping Service %q as it belongs to excluded namespace.", key)
 		return
 	}
 
+	if service.Spec.Type != k8sv1.ServiceTypeClusterIP || service.Spec.ClusterIP == "None" {
```

Comment:
I'm afraid no. Because when handleUpdateService is called for regular services, along with this check, evaluateService we also validate doesIPTablesChainExist etc which is not required at all.

Reply:
ahh I missed the rest of the function - it was unexpanded in Github 😄  How about we put `service.Spec.Type != k8sv1.ServiceTypeClusterIP || service.Spec.ClusterIP == "None"  ` into an `isServiceSupported` function?

Reply:
Right, I was also thinking on the same lines. Thanks @tpantelis for reviewing.